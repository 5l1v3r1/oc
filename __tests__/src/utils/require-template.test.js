const requireTemplate = require('../../../src/utils/require-template');

describe('requireTemplate', () => {
  test('expect type of `requireTemplate` to be function', () => {
    expect(typeof requireTemplate).toBe('function');
  });

  describe('valid', () => {
    const scenarios = [
      { name: 'oc-template-handlebars' },
      { name: 'oc-template-jade' }
    ];

    scenarios.forEach(({ name }) => {
      test(name, () => {
        const template = requireTemplate(name);

        ['compile', 'getCompiledTemplate', 'getInfo', 'render']
          .forEach((method) => {
            expect(template).toHaveProperty(method);
          });
        expect(template).toMatchSnapshot();
      });
    });
  });

  describe('not valid', () => {
    const scenarios = [
      { name: 'lodash' },
      { name: 'oc-invalid-template' }
    ];

    scenarios.forEach(({ name }) => {
      test(name, () => {
        const sut = () => {
          requireTemplate(name);
        };

        expect(sut).toThrow();
        expect(sut).toThrowErrorMatchingSnapshot();
      });
    });
  });
});
