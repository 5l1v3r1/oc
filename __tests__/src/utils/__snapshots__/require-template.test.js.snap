// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`requireTemplate not valid lodash 1`] = `"Error requiring oc-template: \\"lodash\\" is not a valid oc-template"`;

exports[`requireTemplate not valid oc-invalid-template 1`] = `"Error requiring oc-template: \\"oc-invalid-template\\" not found"`;

exports[`requireTemplate valid oc-template-handlebars 1`] = `
Object {
  "compile": [Function],
  "getCompiledTemplate": [Function],
  "getInfo": [Function],
  "render": [Function],
}
`;

exports[`requireTemplate valid oc-template-jade 1`] = `
Object {
  "compile": [Function],
  "getCompiledTemplate": [Function],
  "getInfo": [Function],
  "render": [Function],
}
`;
